packages/flutter_test/lib/src/widget_tester.dart

```diff
@@ -183,6 +202,100 @@ void testWidgets(
   }
 }
 
+WidgetTesterCallback _wrapWithLeakTracking(
+  String description,
+  WidgetTesterCallback callback,
+  LeakTesting? leakTesting,
+) {
+  // This cannot be done just once, because, if a test file starts with a group,
+  // the tear down will happen after group, not after all tests.
+  tearDown(() async {
+    await _mayBeFinalizeLeakTracking();
+  });
+
+  Future<void> wrappedCallBack(WidgetTester tester) async {
+    final LeakTesting settings = leakTesting ?? LeakTesting.settings;
+
+    if (settings.ignore) {
+      await callback(tester);
+      return;
+    }
+
+    if (!_isPlatformSupported) {
+      _mayBePrintPlatformWarning();
+      await callback(tester);
+      return;
+    }
+
+    final PhaseSettings phase = PhaseSettings(
+      name: description,
+      leakDiagnosticConfig: settings.leakDiagnosticConfig,
+      ignoredLeaks: settings.ignoredLeaks,
+      baselining: settings.baselining,
+      ignoreLeaks: settings.ignore,
+    );
+
+    if (!LeakTracking.isStarted) {
+      _setUpLeakTracking();
+    }
+
+    LeakTracking.phase = phase;
+    await callback(tester);
+    LeakTracking.phase = const PhaseSettings.ignored();
+  }
+
+  return wrappedCallBack;
+}
+
+bool _notSupportedWarningPrinted = false;
+bool get _isPlatformSupported => !kIsWeb;
+void _mayBePrintPlatformWarning() {
+  if (!LeakTracking.warnForUnsupportedPlatforms || _isPlatformSupported || _notSupportedWarningPrinted) {
+    return;
+  }
+  _notSupportedWarningPrinted = true;
+  debugPrint(
+    'Leak tracking is not supported on this platform.\n'
+    'To turn off this message, set `LeakTracking.warnForNotSupportedPlatforms` to false.',
+  );
+}
+
+void _dispatchFlutterEventToLeakTracker(ObjectEvent event) {
+  return LeakTracking.dispatchObjectEvent(event.toMap());
+}
+
+void _setUpLeakTracking() {
+  LeakTracking.phase = const PhaseSettings.ignored();
+  LeakTracking.start(config: LeakTrackingConfig.passive());
+  MemoryAllocations.instance.addListener(_dispatchFlutterEventToLeakTracker);
+}
+
+int _plannedTests = 0;
+int _executedTests = 0;
+Future<void> _mayBeFinalizeLeakTracking() async {
+  if (_plannedTests > _executedTests) {
+    return;
+  }
+  MemoryAllocations.instance.removeListener(_dispatchFlutterEventToLeakTracker);
+
+  LeakTracking.declareNotDisposedObjectsAsLeaks();
+  await forceGC(fullGcCycles: defaultNumberOfGcCycles);
+  final Leaks leaks = await LeakTracking.collectLeaks();
+  LeakTracking.stop();
+
+  if (leaks.total == 0) {
+    return;
+  }
+  experimentalCollectedLeaksReporter(leaks);
+}
+
+/// Hendler for memory leaks found by `testWidgets`.
+///
+/// Set it to analyse the leaks programmatically.
+/// The handler is invoked on tear down of the test run.
+/// The default reporter fails in case of found leaks.
+LeaksCallback experimentalCollectedLeaksReporter = (Leaks leaks) => expect(leaks, isLeakFree);
```

Comment:
Added comment.

I do not expect people to use this member. It is needed just to test `testWidgets`. It is similar to `reportTestException` that is used in flutter_test/test/widget_tester_test.dart.

Making it setting in experimentalLeakTesting will add significant complexity to the code of testWidgets, because the reporter will need to change test by test, after tests are completed, requiring to store the handlers in a global variable.

Reply:
If this is only meant for internal testing of flutter_test it should at the very least be marked `@visibleForTesting`. But then again, I would encourage you to find a way to do that without adding more stuff to the global name space because ultimately that still is public API.